src/main/java/io/github/moulberry/notenoughupdates/util/Calculator.java

```diff
@@ -19,19 +19,44 @@
 
 package io.github.moulberry.notenoughupdates.util;
 
+import io.github.moulberry.notenoughupdates.NotEnoughUpdates;
+
 import java.math.BigDecimal;
 import java.math.RoundingMode;
+import java.text.DecimalFormat;
 import java.util.ArrayDeque;
 import java.util.ArrayList;
 import java.util.Deque;
 import java.util.List;
 import java.util.NoSuchElementException;
 
 public class Calculator {
+
+	static String lastInput = "";
+	static String lastResult = null;
+
 	public static BigDecimal calculate(String source) throws CalculatorException {
 		return evaluate(shuntingYard(lex(source)));
 	}
 
+	public static String calculateInSearchBar(String input) {
+		int calculationMode = NotEnoughUpdates.INSTANCE.config.misc.calculationMode;
+		if (input.isEmpty() || calculationMode == 0 || (calculationMode == 1 && !input.startsWith("!"))) return null;
+		input = calculationMode == 1 ? input.substring(1) : input;
+
+		if (!lastInput.equals(input)) {
+			lastInput = input;
+			try {
+				BigDecimal calculate = calculate(input);
+				lastResult = new DecimalFormat("#,##0.##").format(calculate);
+			} catch (CalculatorException ignored) {
+				lastResult = null;
+			}
+		}
+
+		return lastResult;
+	}
+
```

Comment:
I would prefer to keep Calculator.java a pure calculator, untainted by minecraft business logic; therefore id prefer if wed move this to another class (perhaps to the callsite, or to another place)

Reply:
Actually a good idea. I prefer not having the fields in there too. Ill move them.